### PR TITLE
JAMES-3353 Document urn:apache:james:params:jmap:mail:quota JMAP exte…

### PR DESCRIPTION
**Pull request by Arsnael**
…nsion

### CODE REVIEW

File: server/protocols/jmap-rfc-8621/doc/specs/spec/quotas/quotas.mdown [hunk 1]
```diff
@@ -0,0 +1,103 @@
+# Introduction
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/quotas/quotas.mdown:1]: please add this into into.mdown file and move it in the mail subfolder

File: server/protocols/jmap-rfc-8621/doc/specs/spec/quotas/quotas.mdown [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+JMAP ([@!RFC8620] – JSON Meta Application Protocol) is a generic protocol for synchronising data, such as mail, calendars or contacts, between a client and a server. It is optimised for mobile and web environments, and aims to provide a consistent interface to different data types.
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/quotas/quotas.mdown:3]: Drop this intro line, it do not seem to add specific information IMO

File: server/protocols/jmap-rfc-8621/doc/specs/spec/quotas/quotas.mdown [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+This specification defines a data model for handling mail quotas over JMAP, allowing you to read and explain quota information.
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/quotas/quotas.mdown:5]: I think this very spec is only for reading quota.

File: server/protocols/jmap-rfc-8621/doc/specs/spec/quotas/quotas.mdown [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
+"OPTIONAL" in this document are to be interpreted as described in BCP
+14 [@!RFC2119] [@!RFC8174] when, and only when, they appear in all
+capitals, as shown here.
+
+Type signatures, examples and property descriptions in this document follow the conventions established in section 1.1 of [@!RFC8620]. Data types defined in the core specification are also used in this document.
+
+Servers MUST support all properties specified for the new data types defined in this document.
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/quotas/quotas.mdown:21]: IMO this do not add value

File: server/protocols/jmap-rfc-8621/doc/specs/spec/quotas/quotas.mdown [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+## Addition to the capabilities object
+
+The capabilities object is returned as part of the JMAP Session object; see [@!RFC8620], section 2.
+
+This document defines one additional capability URI.
+
+### urn:apache:james:params:jmap:mail:quota
+
+This represents support for the Quota data type and associated API methods. Servers supporting this specification MUST add this property to the capabilities object.
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/quotas/quotas.mdown:37]: State that it is a James specific capability

File: server/protocols/jmap-rfc-8621/doc/specs/spec/quotas/quotas.mdown [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    "quotas":{
+      "quotaRoot":{
+        "Message":{"used":18,"max":42},
+        "Storage":{"used":12,"max":74}
+      },
+      "quotaRoot2@localhost":{
+        "Message":{"used":14,"max":43},
+        "Storage":{"used":19"max":64}
+      }
+    }
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/quotas/quotas.mdown:82]: Please put the full Mailbox object

File: server/protocols/jmap-rfc-8621/doc/specs/spec/quotas/quotas.mdown [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+# Security considerations
+
+All security considerations of JMAP ([@!RFC8620]) apply to this specification.
+
+# IANA Considerations
+
+## JMAP Capability Registration for "quota"
+
+IANA will register the "quota" Apache James Capability as follows:
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/quotas/quotas.mdown:92]: Will it?

File: server/protocols/jmap-rfc-8621/doc/specs/spec/quotas/quotas.mdown (hunk 1 above)
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/quotas/quotas.mdown:1]: No implemented annotation?

**Comment by Arsnael** [comment]
> why not simply linking to the draft instead?

Because the draft being written is already completely different from what we use in James (quotas not being part of mailbox object anymore for example) so I'm not sure we should link to it

**Comment by chibenwa** [comment]
`so I'm not sure we should link to it` yet

**Comment by rouazana** [comment]
> Because the draft being written is already completely different from what we use in James (quotas not being part of mailbox object anymore for example) so I'm not sure we should link to it

Maybe I missed something, but why do we implemented (I mean in the new implementation) something that is different from the draft?

**Comment by chibenwa** [comment]
> Maybe I missed something, but why do we implemented (I mean in the new implementation) something that is different from the draft?

From what I remember we don't want to re-implement a RFC that is far from being finalized.

**Comment by Arsnael** [comment]
We just did a port so far, as long as the RFC is not finalized it is probably better to wait indeed

**Comment by rouazana** [comment]
> We just did a port so far, as long as the RFC is not finalized it is probably better to wait indeed

Not sure it's the best strategy. When submitting an RFC, it is often nice to be able to show a working implementation.

**Comment by Arsnael** [comment]
> Not sure it's the best strategy. When submitting an RFC, it is often nice to be able to show a working implementation.

Maybe, but the new draft is very different from this. That would require some work. And will require extra work to readapt to the finalized version of it later as well. So question is: do we want to pay the cost?

**Comment by chibenwa** [comment]
Build output lost

test this please

**Comment by chibenwa** [comment]
If there is no objections, I will merge this as it describes todays behaviour.

We will implement Quota draft as soon as a consensus emerge within the IETF working group.

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**